### PR TITLE
fix(merge translation): fix merge translation exception with new translation

### PR DESCRIPTION
**Pull request by aeng**
https://zanata.atlassian.net/browse/ZNTA-1013

### CODE REVIEW

File: docs/release-notes.md [hunk 1]
```diff
@@ -1,6 +1,7 @@
 ## 3.8.3
 <h5>Bug fixes</h5>
 * [ZNTA-953](https://zanata.atlassian.net/browse/ZNTA-953) - Fix delete version
+* [ZNTA-1013](https://zanata.atlassian.net/browse/ZNTA-1013) - Fix Merge translation exception
```
Comment (carlosmunoz) [inline docs/release-notes.md]: would this be 3.8.4?

File: zanata-war/src/main/java/org/zanata/service/impl/MergeTranslationsServiceImpl.java [hunk 2]
```diff
@@ -231,6 +238,27 @@ private Integer mergeTranslations(
...
+                                        targetVersionId,
+                                        targetTf.getDocument().getId(),
+                                        targetTf.getId(),
+                                        updatedTarget.getLocale().getLocaleId(),
+                                        updatedTarget.getId(),
+                                        updatedTarget.getState(),
+                                        entry.getValue());
+
+                        Events.instance().raiseTransactionSuccessEvent(
+                                TextFlowTargetStateEvent.EVENT_NAME, event);
```
Comment (carlosmunoz) [inline zanata-war/src/main/java/org/zanata/service/impl/MergeTranslationsServiceImpl.java]: I was thinking more of raising one big event with the information of all the changed textflow targets. That way we gain some performance improvements.
This is no worse than what's happening at the moment though, which is a single event per updated text flow. But we can improve it.

Comment (aeng) [inline zanata-war/src/main/java/org/zanata/service/impl/MergeTranslationsServiceImpl.java]: That will be more change - new Event type to take in list of informations which I tried to minimise.

Comment (carlosmunoz) [inline zanata-war/src/main/java/org/zanata/service/impl/MergeTranslationsServiceImpl.java]: OK, Lets then at least put a TODO on it so we can refactor in the future.

Comment (aeng) [inline zanata-war/src/main/java/org/zanata/service/impl/MergeTranslationsServiceImpl.java]: Done.

**Comment by carlosmunoz** [comment]
:+1:

**Comment by djansen-redhat** [comment]
:white_check_mark: I *believe* this works as expected

**Pull request merged**